cdist/conf/type/__start_on_boot/explorer/state

```diff
@@ -24,47 +24,51 @@
 
 os=$("$__explorer/os")
 runlevel=$("$__explorer/runlevel")
+init=$("$__explorer/init")
 target_runlevel="$(cat "$__object/parameter/target_runlevel")"
 name="$__object_id"
 
-case "$os" in
-    archlinux)
-        state=$(systemctl is-enabled "$name" >/dev/null 2>&1 \
-            && echo present \
-            || echo absent)
-    ;;
+if [ "$init" == 'systemd' ]; then
```

Comment:
I agree with this approach - we should indeed use the "correct" init system, independently of which distro it actually is